src/frontend/Contexts/CategoriesContext.js

```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+import { createContext, useContext, useState, useEffect } from "react";
+
+const CategoriesContext = createContext();
+
+const CategoriesProvider = ({ children }) => {
+  const [categories, setCategories] = useState([]);
+  useEffect(() => {
+    (async () => {
```

Comment:
you can use separate function so that your code looks more cleaner

Reply:
I used IIFE